src/core/thread/mle_router.cpp

```diff
@@ -423,17 +423,14 @@ ThreadError MleRouter::ResetAdvertiseInterval(void)
 {
     uint32_t advertiseDelay;
 
-    VerifyOrExit(mAdvertiseInterval != kAdvertiseIntervalMin || !mAdvertiseTimer.IsRunning(), ;);
```

Comment:
I do not get that the timer may be never fired, and alway thinks about reset the timer while reading the code. Thanks for the comments! Closing it soon.